src/builder.rs

```diff
@@ -1464,6 +1464,102 @@ impl Builder {
 
         BasicValueEnum::new(value)
     }
+
+    /// Builds an atomicrmw instruction. It allows you to atomically modify memory.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use inkwell::context::Context;
+    /// use inkwell::{AddressSpace, AtomicOrdering, AtomicRMWBinOp};
+    /// let context = Context::create();
+    /// let module = context.create_module("rmw");
+    /// let void_type = context.void_type();
+    /// let i32_type = context.i32_type();
+    /// let i32_seven = i32_type.const_int(7, false);
+    /// let i32_ptr_type = i32_type.ptr_type(AddressSpace::Generic);
+    /// let fn_type = void_type.fn_type(&[i32_ptr_type.into()], false);
+    /// let fn_value = module.add_function("rmw", fn_type, None);
+    /// let entry = fn_value.append_basic_block("entry");
+    /// let i32_ptr_param = fn_value.get_first_param().unwrap().into_pointer_value();
+    /// let builder = context.create_builder();
+    /// builder.position_at_end(&entry);
+    /// builder.build_atomicrmw(AtomicRMWBinOp::Add, i32_ptr_param, i32_seven, AtomicOrdering::Unordered);
+    /// builder.build_return(None);
+    /// ```
+    // https://llvm.org/docs/LangRef.html#atomicrmw-instruction
+    pub fn build_atomicrmw(&self, op: AtomicRMWBinOp, ptr: PointerValue, value: IntValue, ordering: AtomicOrdering) -> Result<InstructionValue, ()> {
+        // "The type of ‘<value>’ must be an integer type whose bit width is a power of two greater than or equal to eight and less than or equal to a target-specific size limit. The type of the ‘<pointer>’ operand must be a pointer to that type." -- https://releases.llvm.org/3.6.2/docs/LangRef.html#atomicrmw-instruction
+        // Newer LLVM's (9+) support additional FAdd and FSub operations as well as xchg on floating point types.
+        if value.get_type().get_bit_width() < 8 ||
+           !value.get_type().get_bit_width().is_power_of_two() {
+            return Err(());
```

Comment:
Can we provide users with helpful static string error messages? Like `"Value must be a power of two"`?

Reply:
Done!